prExplorer/Application/Products/Delete.cs

```diff
@@ -14,17 +17,34 @@ public class Command : IRequest<Result<Unit>>
         public class Handler : IRequestHandler<Command, Result<Unit>>
         {
             private readonly DataContext _context;
-            public Handler(DataContext context)
+            private readonly IUserAccessor _userAccessor;
+            public Handler(DataContext context, IUserAccessor userAccessor)
             {
+                _userAccessor = userAccessor;
                 _context = context;
             }
             public async Task<Result<Unit>> Handle(Command request, CancellationToken cancellationToken)
             {
-                var product = await _context.Products.FindAsync(request.Id);
+
+                var user = await _context.Users.FirstOrDefaultAsync(x => x.Id == _userAccessor.GetUserId());
+
+                if (user == null)
+                {
+                    // Eğer kullanıcı bulunamazsa hata işleme.
```

Comment:
I suggest you to always use English is code, especially in the comments. What if, one day, you need an additional developer to help you who doesn't know Turkish?